### ISSUE TITLE
Loosen string comparison

### ISSUE THREAD

**Issue by isidornygren** [JakeSidSmith/tsassert#7]
As currently, it will fail if the types have a different amount of whitespaces, e.g if the checked type is `{ name: string, value: number }` it will fail the check if the defined type is `{name:string,value:number}` which imo shouldn't fail, as it would be nice to be able to format the types in the comment nicer. 
Especially if multiline type comments for complex types would work in the future, some kind of whitespace remover would definitely be needed.

**Comment by JakeSidSmith** [comment]
I'm reluctant to loosen the string comparison, as there is potential for types that do not match to pass, however, to make it easier to see where whitespace differs I have added/improved the output to highlight differences:

![diff](https://user-images.githubusercontent.com/5850625/76127895-dba97500-5ffa-11ea-8d1e-0accb96c1077.png)

Would you say that this resolves this issue?

**Comment by JakeSidSmith** [comment]
Also, I just quite like that this is strict. Strict is what I always intended.

**Comment by isidornygren** [comment]
That's fine I guess.
What would happen if the type was really complex then? Would you have to write everything in one line?
```typescript
const something = someFunc(); // { value: { inner: number, innerString: string}, someOtherValue: string, moreValue: { date: Date, isFuture: boolean, hasKetchup: boolean } }
```

**Comment by JakeSidSmith** [comment]
Yup. But you'd generally be pulling those inner parts out into their own types, so the assertions shouldn't really get that large, and instead end up with `{ value: InnerThing, someOtherValue: Something }` or even just `AllMyValues` possibly with generics.
